rice-middleware/impl/src/test/java/org/kuali/rice/kew/actions/CancelActionTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2006-2017 The Kuali Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.opensource.org/licenses/ecl2.php
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kuali.rice.kew.actions;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kuali.rice.kew.actionrequest.ActionRequestValue;
+import org.kuali.rice.kew.api.action.ActionType;
+import org.kuali.rice.kew.api.document.DocumentStatus;
+import org.kuali.rice.kew.routeheader.DocumentRouteHeaderValue;
+import org.kuali.rice.kim.api.identity.principal.PrincipalContract;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CancelActionTest {
+
+    private static final String PRINCIPAL_ID = "01234567";
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS) private DocumentRouteHeaderValue routeHeaderValue;
+
+    @Mock private  PrincipalContract principalContract;
+
+    @Mock private ActionRequestValue actionRequestValue;
+
+    private CancelAction cancelAction;
+
+    private List<ActionRequestValue> actionRequestValues;
+
+    @Before public void setUp() {
```

Comment:
I was following conventions from other tests in foundation Rice, but will move all annotations to their own line, as I prefer.